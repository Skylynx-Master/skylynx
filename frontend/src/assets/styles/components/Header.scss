* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

header {
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: center;
  background-color: #47cfac;
}

.Header-logo {
  width: 120px;
  height: 40%;
}

.Header-links {
  list-style: none;
}

.Header-links li {
  display: inline-block;
  padding: 0px 20px 0px 5px;
  color: #3f3f44;
  font-family: "Signika", sans-serif;
}

.Header-links li:hover {
  color: #f7f7f7;
  cursor: pointer;
}
