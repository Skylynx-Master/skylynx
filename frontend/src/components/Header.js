import React from "react";
import "../assets/styles/components/Header.scss";

function Header() {
  return (
    <header className="Header">
      <img className="Header-logo"></img>
      <nav className="Header-Navbar">
        <ul className="Header-link">
          <a>
            <li>Hola header</li>
          </a>
          <li>Hola Item</li>
          <li>Hola Favoritos</li>
          <li>Hola Favoritos</li>
        </ul>
      </nav>
    </header>
  );
}

export default Header;
