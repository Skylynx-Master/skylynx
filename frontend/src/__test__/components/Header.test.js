import React from "react";
import Header from "../../components/Header";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import HamburgerMenu from "../../components/HamburgerMenu";

describe("Header Component Testing", () => {
  const header = mount(
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );

  const hamburgerMenu = mount(
    <BrowserRouter>
      <HamburgerMenu />
    </BrowserRouter>
  );

  test("Header Render", () => {
    expect(header.length).toEqual(1);
  });

  test("Header Logo Render", () => {
    expect(header.find(".Header-logo").length).toEqual(1);
  });

  test("Link title Search/Buscar", () => {
    expect(header.find(".Header-icons_search").text()).toEqual("Buscar");
  });
  test("Link title Favorites/Favoritos", () => {
    expect(header.find(".Header-icons_favorites").text()).toEqual("Favoritos");
  });

  test("Link title Create Post/Crear Publicacion", () => {
    expect(header.find(".Header-icons_createpost").text()).toEqual(
      "Crear PublicaciÃ³n"
    );
  });

  test("Header Icons and Links Render", () => {
    expect(header.find(".Header-icons").length).toEqual(8);
  });

  test("Hamburger Menu Availability", () => {
    expect(hamburgerMenu.length).toEqual(1);
  });
});

describe("Header Snapshot", () => {
  test("Test UI Header component", () => {
    const snapheader = create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(snapheader.toJSON()).toMatchSnapshot();
  });
});
