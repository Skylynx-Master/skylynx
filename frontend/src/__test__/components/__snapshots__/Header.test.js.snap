// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Test UI Header component 1`] = `
<header
  className="Header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Logo LaHause"
      className="Header-logo"
      src=""
    />
  </a>
  <nav>
    <ul
      className="Header-links"
    >
      <a
        className="Header-icons"
        href="/filters"
        onClick={[Function]}
      >
        <img
          alt="search icon"
          src=""
        />
        <li
          className="Header-icons_search"
        >
          Buscar
        </li>
      </a>
      <a
        className="Header-icons"
      >
        <img
          alt="favorites icon"
          src=""
        />
        <li
          className="Header-icons_favorites"
        >
          Favoritos
        </li>
      </a>
      <a
        className="Header-icons"
        href="/create-post"
        onClick={[Function]}
      >
        <img
          alt="create post icon"
          src=""
        />
        <li
          className="Header-icons_createpost"
        >
          Crear Publicación
        </li>
      </a>
      <a
        className="Header-icons"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="home icon"
          src=""
        />
        <li>
          Inicio
        </li>
      </a>
    </ul>
    <div
      className="sc-AxirZ gfqYih"
      onClick={[Function]}
      open={false}
    >
      <div />
      <div />
      <div />
    </div>
    <ul
      className="sc-AxjAm buYwmk"
      open={false}
    >
      <a
        href="/filters"
        onClick={[Function]}
      >
        <li>
          Buscar
        </li>
      </a>
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <li>
          Favoritos
        </li>
      </a>
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <li>
          Crear Publicación
        </li>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <li>
          Inicio
        </li>
      </a>
    </ul>
  </nav>
</header>
`;
